lib/valueflow.cpp

```diff
@@ -6273,9 +6270,6 @@ struct SimpleConditionHandler : ConditionHandler {
         if (!conds.empty())
             return conds;
 
-        Condition cond;
-        ValueFlow::Value true_value;
```

Comment:
These are not detected since the class has another constructor with unknown implementation: https://trac.cppcheck.net/ticket/11216

Reply:
That other constructor has no side effects either but that is not visible. It might be helpful to configure this via a library: https://trac.cppcheck.net/ticket/11217